controllers/vmware/test/controllers_suite_test.go

```diff
@@ -102,6 +104,16 @@ func getTestEnv() (*envtest.Environment, *rest.Config) {
 }
 
 func getManager(cfg *rest.Config, networkProvider string) manager.Manager {
+	contentFmt := `username: '%s'
+	password: '%s'
+	`
+	tmpFile, err := os.CreateTemp("", "creds")
+	Expect(err).NotTo(HaveOccurred())
+
+	content := fmt.Sprintf(contentFmt, cfg.Username, cfg.Password)
+	_, err = tmpFile.Write([]byte(content))
+	Expect(err).NotTo(HaveOccurred())
```

Comment:
Curious as to how it was working before, just skeptical if we are changing any default functionality

Reply:
We introduced the Watch in the main function which expects the creds file to exist. Before this we were using the credential values from the env variable. We're still doing similar thing but reading it through the file initially for the Watch to function. Thus we are simulating the existence of the file in the tests (just like in the Unit tests) so that the go-routines don't report failures on initialization.

Reply:
So this means that we will need the file to be present always? Or am I getting that incorrectly?

Reply:
Yes for the e2e test to succeed (otherwise it will give error from this point https://github.com/kubernetes-sigs/cluster-api-provider-vsphere/blob/2ce86d348006a3893931853751283e479dc1b521/pkg/manager/options.go#L129). We have covered the failure case in unit test.